server/monitor/src/main/java/org/apache/accumulo/monitor/rest/tables/TablesResource.java

```diff
@@ -0,0 +1,269 @@
+          } else if (tableName.startsWith(name.namespace + ".")) {
+            name.addTable(new TableInformation(tableName, tableId, tableManager.getTableState(tableId).name()));
+          }
+        }
+        tables.add(new TableInformation(tableName, tableId, tableManager.getTableState(tableId).name()));
+      }
+    }
+
+    return tableNamespace;
+  }
+
+  /**
+   * Generates a list of all the tables
+   *
+   * @return list with all tables
+   */
+  @GET
+  public TablesList getTables() {
+    return generateTables("*");
+  }
+
+  /**
+   * Generates a list with the selected namespace
+   *
+   * @param namespace
+   *          Namespace to filter tables
+   * @return list with selected tables
+   */
+  @GET
+  @Path("namespace/{namespace}")
+  public TablesList getTable(@PathParam("namespace") String namespace) {
+    return generateTables(namespace);
+  }
+
+  /**
+   * Generates a list with the list of namespaces
+   *
+   * @param namespaceList
+   *          List of namespaces separated by a comma
+   * @return list with selected tables
+   */
+  @GET
+  @Path("namespaces/{namespaces}")
+  public TablesList getTableWithNamespace(@PathParam("namespaces") String namespaceList) {
+    SortedMap<String,String> namespaces = Namespaces.getNameToIdMap(Monitor.getContext().getInstance());
+
+    TablesList tableNamespace = new TablesList();
+    /*
+     * Add the tables that have the selected namespace Asterisk = All namespaces Hyphen = Default namespace
+     */
+    for (String namespace : namespaceList.split(",")) {
+      for (String key : namespaces.keySet()) {
+        if (namespace.equals("*") || namespace.equals(key) || (key.equals("") && namespace.equals("-"))) {
+          tableNamespace.addTable(new TableNamespace(key));
+        }
+      }
+    }
+
+    return generateTables(tableNamespace);
+  }
+
+  /**
+   * Generates a list of participating tservers for a table
+   *
+   * @param tableId
+   *          Table ID to find participating tservers
+   * @return List of participating tservers
+   */
+  @Path("{tableId}")
+  @GET
+  public TabletServers getParticipatingTabletServers(@PathParam("tableId") String tableId) throws Exception {
+    Instance instance = Monitor.getContext().getInstance();
+
+    TreeSet<String> locs = new TreeSet<>();
+    if (RootTable.ID.equals(tableId)) {
+      locs.add(instance.getRootTabletLocation());
+    } else {
+      String systemTableName = MetadataTable.ID.equals(tableId) ? RootTable.NAME : MetadataTable.NAME;
+      MetaDataTableScanner scanner = new MetaDataTableScanner(Monitor.getContext(), new Range(KeyExtent.getMetadataEntry(tableId, new Text()),
+          KeyExtent.getMetadataEntry(tableId, null)), systemTableName);
+
+      while (scanner.hasNext()) {
+        TabletLocationState state = scanner.next();
+        if (state.current != null) {
+          try {
+            locs.add(state.current.hostPort());
+          } catch (Exception ex) {}
```

Comment:
If you don't have any exception handling on the server, it would be better to at least throw exception back to the client.  It is ugly when it happens but at least it won't be hidden.